src/cares_wrap.cc

```diff
@@ -2153,6 +2153,33 @@ void SetServers(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(err);
 }
 
+void SetLocalAddress(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  ChannelWrap* channel;
+  ASSIGN_OR_RETURN_UNWRAP(&channel, args.Holder());
+
+  CHECK(args[0]->IsString());
+
+  Isolate* isolate = args.GetIsolate();
+  node::Utf8Value ip(isolate, args[0]);
+
+  unsigned char addr[sizeof(struct in6_addr)];
+
+  if (uv_inet_pton(AF_INET, *ip, &addr) == 0) {
+    unsigned int ip4 = (addr[0] << 24) +
```

Comment:
There's a pretty subtle case of implementation-defined behavior (IB) here: `addr[0]` gets promoted to `int` (because `24` is an int) and is then left-shifted.

If `addr[0] >= 0x80`, that ends up shifting into the sign bit and that's IB.

Replace `24` with `24u` and all is well.

That said, you should be able to use `cares_get_32bit()` here.

Looks like we have more instances of that. I've filed https://github.com/nodejs/node/issues/34827.